arcgis-ios-sdk-samples/Edit data/Update geometries (feature service)/EditGeometryViewController.swift

```diff
@@ -64,91 +95,89 @@ class EditGeometryViewController: UIViewController, AGSGeoViewTouchDelegate, AGS
-        // hide the callout
-        self.mapView.callout.dismiss()
-        
-        // add the default geometry
-        let point = self.selectedFeature.geometry as! AGSPoint
-        
-        // instantiate sketch editor with selected feature's geometry
-        self.mapView.sketchEditor = AGSSketchEditor()
-        
-        // enable the sketch editor to start tracking user gesture
-        self.mapView.sketchEditor?.start(with: point)
-        
-        // show the toolbar
-        self.setToolbarVisibility(visible: true)
-        
-        // hide the feature for time being
-        self.featureLayer.setFeature(self.selectedFeature, visible: false)
+        guard let point = selectedFeature.geometry as? AGSPoint else { return }
+        // Hide the callout.
+        mapView.callout.dismiss()
+        
+        // Instantiate sketch editor with selected feature's geometry, and
+        // enable the sketch editor to start tracking user gesture.
+        let sketchEditor = AGSSketchEditor()
+        mapView.sketchEditor = sketchEditor
+        sketchEditor.start(with: point)
+        
+        // Show the toolbar.
+        setToolbarVisibility(visible: true)
+        // Hide the feature for time being.
+        featureLayer.setFeature(selectedFeature, visible: false)
     }
     
     // MARK: - Actions
     
     @IBAction func doneAction() {
-        if let newGeometry = self.mapView.sketchEditor?.geometry {
-            self.selectedFeature.geometry = newGeometry
-            self.featureTable.update(self.selectedFeature) { [weak self] (error) in
+        if let newGeometry = mapView.sketchEditor?.geometry {
+            selectedFeature.geometry = newGeometry
+            featureTable.update(selectedFeature) { [weak self] (error: Error?) in
+                guard let self = self else { return }
                 if let error = error {
-                    self?.presentAlert(error: error)
-                    
-                    // un hide the feature
-                    self?.featureLayer.setFeature(self!.selectedFeature, visible: true)
+                    self.presentAlert(error: error)
+                    // Make the feature visible due to unsuccessful update.
+                    self.featureLayer.setFeature(self.selectedFeature, visible: true)
                 } else {
-                    // apply edits
-                    self?.applyEdits()
+                    // Apply edits.
+                    self.applyEdits()
                 }
             }
         }
-        
-        // hide toolbar
-        self.setToolbarVisibility(visible: false)
-        
-        // disable sketch editor
-        self.mapView.sketchEditor?.stop()
-        
-        // clear sketch editor
-        self.mapView.sketchEditor?.clearGeometry()
+        // Hide toolbar.
+        setToolbarVisibility(visible: false)
+        // Stop and clear sketch editor.
+        mapView.sketchEditor?.stop()
+        mapView.sketchEditor = nil
```

Comment:
Actually from reading another sample, I don't think it's needed to nullify the sketch editor every time. Can just create once in mapView didSet.